incubator/hnc/e2e/tests/delete-anchor-crd_test.go

```diff
@@ -0,0 +1,74 @@
+package test
+
+import (
+	"os/exec"
+	"os"
+	"time"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Delete-anchor-crd", func() {
+	It("should create parent and deletable child, and delete the CRD", func() {
+		GinkgoT().Log(time.Now())
+
+		// create a namespace delete-crd-parent
```

Comment:
You don't need to comment every line, especially if the code is self-explaining. For example, if the code is:

`runcmd("kubectl create ns delete-crd-parent")`

Then the comment

`// Create the namespace delete-crd-parent`

basically just repeats what the code says, and doesn't add any new information. What might be more useful is:

```
// Setup
runcmd("kubectl create ns delete-crd-parent")
runcmd("kubectl hns create delete-crd-child -n delete-crd-parent")

// Test
runcmd("kubectl hns set delete-crd-child -a")

// Verify
...

// Cleanup
...
```

Because now the reader can very quickly look at the part of the test they care about. Then you only need to add _additional_ comments if it's unclear _why_ the code is doing what it's doing.